### PR TITLE
Support datalab connect() to connect a GCE instance created from Deep…

### PR DESCRIPTION
**Pull request by yixinshi**
…learning images

### CODE REVIEW

File: tools/cli/commands/utils.py [hunk 1]
```diff
@@ -68,7 +68,8 @@ class InvalidInstanceException(Exception):
 
     _MESSAGE = (
         'The specified instance, {}, does not appear '
-        'to have been created by the `datalab` tool, and '
+        'to have been created by the `datalab` tool, or '
+        'from any GCE Deeplearning images, and '
```
Comment (qimingj) [inline tools/cli/commands/utils.py]: Do we need the "and"?

File: tools/cli/commands/utils.py [hunk 2]
```diff
@@ -240,22 +241,35 @@ def flatten_metadata(metadata):
     return result
 
 
-def _check_datalab_tag(instance, tags):
+def _check_datalab_tag(instance, status_tags_and_metadata):
```
Comment (qimingj) [inline tools/cli/commands/utils.py]: Also rename the function?

File: tools/cli/commands/utils.py (hunk 1 above)
Comment (yixinshi) [reply tools/cli/commands/utils.py]: I rewrote it.

File: tools/cli/commands/utils.py (hunk 2 above)
Comment (yixinshi) [reply tools/cli/commands/utils.py]: Done

**Comment by yixinshi** [comment]
Thanks, qimingj. PTAL

**Comment by qimingj** [comment]
There is a test failure. I just reran it and see if it is just flaky.

File: tools/cli/commands/utils.py [hunk 3]
```diff
@@ -68,8 +68,9 @@ class InvalidInstanceException(Exception):
 
     _MESSAGE = (
         'The specified instance, {}, does not appear '
-        'to have been created by the `datalab` tool, and '
-        'so cannot be managed by it.')
+        'to have been created by the `datalab` tool, or '
+        'from any GCE Deeplearning images. Therefore it '
+        'so cannot be managed by `datalab` tool.')
```
Comment (ojarjur) [inline tools/cli/commands/utils.py]: The "so " here is redundant.

Comment (yixinshi) [reply tools/cli/commands/utils.py]: Done. Can you grant me the write access to the repo? Thanks!

**Pull request merged**